internal/fs/read_cache_test.go

```diff
@@ -0,0 +1,363 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A collection of tests for a file system where the file cache is enabled
+// without download-file-for-random-read set to True.
+
+package fs_test
+
+import (
+	"io"
+	"math/rand"
+	"os"
+	"path"
+	"reflect"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/util"
+	"github.com/googlecloudplatform/gcsfuse/internal/config"
+	. "github.com/jacobsa/ogletest"
+)
+
+// //////////////////////////////////////////////////////////////////////
+// Boilerplate
+// //////////////////////////////////////////////////////////////////////
+const FileCacheSizeInMb = 10
+
+var CacheLocation = path.Join(os.Getenv("HOME"), "cache-dir")
+
+const DefaultObjectName = "foo.txt"
+const DefaultObjectSizeInMb = 5
+
+type FileCacheTest struct {
+	fsTest
+}
+
+func init() {
+	RegisterTestSuite(&FileCacheTest{})
+}
+
+func (t *FileCacheTest) SetUpTestSuite() {
+	t.serverCfg.ImplicitDirectories = true
+	t.serverCfg.MountConfig = &config.MountConfig{
+		FileCacheConfig: config.FileCacheConfig{
+			MaxSizeInMB:               FileCacheSizeInMb,
+			DownloadFileForRandomRead: false,
+		},
+		CacheLocation: config.CacheLocation(CacheLocation),
+	}
+	t.fsTest.SetUpTestSuite()
+}
+
+func (t *FileCacheTest) TearDown() {
+	t.fsTest.TearDown()
+	_ = os.RemoveAll(CacheLocation)
```

Comment:
Should we Assert the error returned by os.RemoveAll? As it might affect other test-cases?